ansible/roles/invoker/tasks/deploy.yml

```diff
@@ -31,6 +31,19 @@
     linuxOptions: "-v /usr/lib/x86_64-linux-gnu/libapparmor.so.1:/usr/lib/x86_64-linux-gnu/libapparmor.so.1"
   when: whisk_version_name != "mac"
 
+- name: get running invoker information
+  uri: url="http://{{ inventory_hostname }}:{{ docker.port }}/containers/json?filters={{ '{"name":[ "invoker" ]}' | urlencode }}" return_content=yes
+  register: invokerInfo
+
+- name: determine if more than one invokers are running
+  fail: msg="more than one invokers are running"
+  when: "{{ invokerInfo.json|length > 1 }}"
+
+- name: determine if index of invoker is same with index of inventory host
+  fail: msg="invoker index is invalid {{item.Names[0]}} /invoker{{play_hosts.index(inventory_hostname)}}"
+  with_items: "{{ invokerInfo.json }}"
+  when: '"{{ item.Names[0] }}" != "/invoker{{play_hosts.index(inventory_hostname)}}"'
```

Comment:
Same comment as the other, please get rid of as many template strings as possible:

`when: item.Names[0] != "/invoker{{play_hosts.index(inventory_hostname)}}"`

Should work I think.